posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc

```diff
@@ -0,0 +1,118 @@
+
+== Managing your applications
+
+Any application in your Eclipse workspace that is configured with the Liberty Maven or Liberty Gradle plugin is automatically enabled to run with Liberty Tools. Actions are available to manage these apps such as `Start`, `Stop`, `Debug` and `Run Tests`, and you can run these actions from either the Liberty Dashboard or the Eclipse "Run As" menu.
+
+=== Liberty Dashboard
+
+The Liberty Dashboard consolidates the Liberty enabled applications in your workspace into a single view. To open the dashboard, click on Open Liberty UFO icon on the top tray. To run an action, simply right click on an app in the list.
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-dash-menu.png[Liberty Dashboard,width=70%,align="center"]
+
+=== Eclipse Run As menu
+
+Liberty Tools offers the same set of actions in the Eclipse "Run As" menu that appears when you right click on a project in the Project Explorer view.
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-runas-menu.png[Run As menu,width=70%,align="center"]
+
+=== Starting your app
+
+To start your application in dev mode, select either the `Start` or `Start...` actions. The latter will open a `Run Configurations` dialog box where you can add link:https://github.com/OpenLiberty/ci.maven/blob/main/docs/dev.md#additional-parameters[additional parameters] to dev mode or configure the JRE to use when running the app. When a start action is selected, Liberty Tools opens a `Terminal` tab and starts dev mode. 
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-run-config.gif[Run Configurations,width=70%,align="center"]
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-terminal-start.png[Terminal Start,width=70%,align="center"]
+
+NOTE: The `Start in container` action will link:https://github.com/OpenLiberty/ci.maven/blob/main/docs/dev.md#devc-container-mode[start dev mode in a local Docker container]. 
+
+=== Running tests
+
+Tests are important for any application. Liberty Tools makes it easy to run your unit and integration tests by either selecting the `Run tests` action, or simply clicking `Enter` in the terminal. You can even view the results of your latest run by selecting `View test reports`.
+
+=== Debugging your app
+
+The Liberty runtime is configured with port 7777 for debugging by default. However, this requires that port to be available on your system and for you to manually attach a debugger. With Liberty Tools, you can start your app using the "Debug" action which will find an available port and automatically attach the Eclipse debugger to it. 
+
+== Get coding!
+
+=== Jakarta EE 
+
+Code completion for Jakarta EE makes it easy to add classes and methods to your app. Just type `CTRL` + `SPACE` from within any Java file to see a list of available code snippets to choose from. Additionally, diagnostics flag errors and suggest quick-fixes to implement on the spot.
```

Comment:
I feel like I usually see this as `Ctrl + Space`  or `CTRL +Space`  but don't usually see SPACE in caps.

Reply:
ok - changed